sqlglot/dialects.py

```diff
@@ -461,16 +630,16 @@ def _unix_to_time(self, expression):
 Hive.functions = {
     "APPROX_COUNT_DISTINCT": exp.ApproxDistinct.from_arg_list,
     "COLLECT_LIST": exp.ArrayAgg.from_arg_list,
-    "DATE_ADD": lambda args: exp.TsOrDsAdd(
```

Comment:
this is fishy

Reply:
`DATE_ADD` is really confusing. For example Hive accepts a date str and returns a date string. Presto expects a date or timestamp and returns a type that matches the input. Duckdb technically doesn't have a DATE_ADD but you can use interval +/- and that behaves like presto. Therefore I start switching the implementations to not change the default behavior of `DATE_ADD` but instead have it use the `DATE_ADD` for that vendor directly. In my mind if a user wants a cross-vendor functionality they should use `MixedTypeAdd` and define an output format in order to be flexible on the input and explicit on expected output.

Reply:
i don't think that approach is safe. users write hive sql and expect it to work in presto. thus the intention of DATE_ADD is given a datestr add it. so that's why in the original code, when hive has date_add it really means TSorDSADD. 

i don't believe we should change this behavior as it's not an optimal experience. the goal is for users to write once and run anywhere. it is indeed tricky because hive sql does a lot of implicit stuff.

Reply:
If a user wants to do cross-vendor functionality, they can writing sqlglot syntax MixedTypeAdd, but it should not be required to get transpilation

Reply:
there is no date object in hive and so this works doesn’t it? perhaps the old code was bugged and in this case hive should convert to string first?

Reply:
Hive does have a date type: https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=82706456#LanguageManualTypes-date

So I think the ideal way to handle this would be to convert presto's `DATE_ADD` to a `DATE + INTERVAL` like Duckdb does. Hive's date_add would convert to `MixedTypeAdd` in presto with an output format of `YYYY-MM-DD`. I need to look at this a bit more to see how feasible this is though.